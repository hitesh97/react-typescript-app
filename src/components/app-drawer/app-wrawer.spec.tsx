import * as React from 'react';
import { shallow, mount } from 'enzyme';
import AppDrawer from './app-drawer';
import {createShallow, createMount } from 'material-ui/test-utils';
import { MountOptions } from 'material-ui/test-utils/createMount';
import getClasses from 'material-ui/test-utils/getClasses';

describe('AppDrawer', () => {

    let materialMount: any, classes: any;
    beforeEach(() => {
        materialMount = createShallow();
        classes = getClasses(<AppDrawer />);
    });

    describe('without props', () => {
        it('should match a snapshot', () => {
            const component = materialMount(<AppDrawer />);
            expect(component).toMatchSnapshot();
        });
    });

    describe('With prop Open true', () => {
        it('should match a snapshot', () => {
            const component = materialMount(<AppDrawer open={true} />);
            expect(component).toMatchSnapshot();
            expect(component.props().open).toBe(true);
        });
    });
});
