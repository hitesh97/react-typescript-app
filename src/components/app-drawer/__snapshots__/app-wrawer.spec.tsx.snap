// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDrawer With prop Open true should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Connect(AppDrawer))
    open={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "list": "Connect-AppDrawer--list-2",
      },
      "open": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "list": "Connect-AppDrawer--list-2",
        },
        "open": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;

exports[`AppDrawer without props should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Connect(AppDrawer)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "list": "Connect-AppDrawer--list-1",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "list": "Connect-AppDrawer--list-1",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
